node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -131,39 +145,99 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
+         * @param  {Function}       callback            Standard callback function
+         * @param  {Boolean}        _isRetryAttempt    `true` if we attempted to load this page previously
+         * @param  {Object}         jqXHR               jQuery XHR from the failed load attempt
+         */
+        var handlePageLoadError = function(page, callback, _isRetryAttempt, jqXHR) {
+            // If the user has been reading a document for a while, it's possible that the signature expired
+            // If we could not load the page due to a 401 (access denied) and it's the first time we tried loading this
+            // particular page, we refresh the signature and try again
+            if (jqXHR.status === 401 && !_isRetryAttempt) {
+                refreshSignature(function(err) {
+                    if (err) {
+                        callback(err);
+                    }
+
+                    // Try loading the page again
+                    loadPage(page, callback, true);
+                });
+            }
+
+            // For all other error responses, we pass the error back up the call stack
+            callback({'code': jqXHR.status, 'msg': jqXHR.responseText});
+        };
+
+        /**
+         * Load a document page and append it to the document viewer
+         *
+         * @param  {Object}         page                Page object representing the page that needs to be loaded and rendered
+         * @param  {Function}       [callback]          Standard callback function
+         * @param  {Object}         [callback.err]      Error object containing error code and error message
+         * @param  {Boolean}        [_isRetryAttempt]   `true` if we attempted to load this page previously
+         */
+        var loadPage = function(page, callback, _isRetryAttempt) {
+            // Set a default callback function in case no callback function has been provided
+            callback = callback || function() {};
+            // Indicates whether or not we attempted to load this page previously
+            _isRetryAttempt = _isRetryAttempt || false;
+
+            // Don't reload the page if all of its components have already been loaded
+            if (page.$el && page.$html && page.$text) {
+                return callback();
+            }
+
+            // Indicate that a page is being loaded, so no other pages are loaded at the same time
+            isLoadingPage = true;
+
+            // Keep track of the last page in the document that was loaded
+            lastPageLoaded = page.pageNumber;
+
+            // Set up the HTML markup for the page
+            prepareForPageLoading(page);
+
+            // Kick off the request for the page's text content (for screen readers).
+            // Load the text first so we can rely on the HTML load to catch signature
+            // expiry and initiate a recovery.
+            if (!page.$text) {
```

Comment:
I think we want the HTML page to be loaded first, as that will refresh the signature.

```
loadPageHtml(page, function(jqXHR) {
  ...
  loadPageText(page);
  ...
  callback();
```